wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/component/editor/EditorComponentToWorkspaceApplier.java

```diff
@@ -68,35 +69,42 @@ public void apply(
 
     String editorComponentAlias = editorComponent.getAlias();
     String editorId = editorComponent.getId();
+    String registryUrl = editorComponent.getRegistryUrl();
+    String reference = editorComponent.getReference();
     String memoryLimit = editorComponent.getMemoryLimit();
 
-    workspaceConfig.getAttributes().put(WORKSPACE_TOOLING_EDITOR_ATTRIBUTE, editorId);
-
     if (editorComponentAlias != null) {
       workspaceConfig
           .getAttributes()
           .put(EDITOR_COMPONENT_ALIAS_WORKSPACE_ATTRIBUTE, editorComponentAlias);
     }
 
-    final ExtendedPluginFQN fqn;
-    try {
-      fqn = fqnParser.parsePluginFQN(editorId);
-    } catch (InfrastructureException e) {
-      throw new DevfileException(e.getMessage(), e);
-    }
-    if (memoryLimit != null) {
+    if (!isNullOrEmpty(reference)) {
+      workspaceConfig.getAttributes().put(WORKSPACE_TOOLING_EDITOR_ATTRIBUTE, reference);
+    } else {
+      String compositeId = registryUrl != null ? registryUrl + "#" + editorId : editorId;
```

Comment:
Why not make this a method of the `PluginFQNParser` or maybe rather `PluginFQN`? Could be combined with the logic of `parsePluginFQN` below and result in a simpler flow here.